src/wesnoth.cpp

```diff
@@ -1099,24 +1096,37 @@ int main(int argc, char** argv)
 			return pos == std::string::npos ? arg == sw : arg.substr(0, pos) == sw;
 		};
 
-		if(wincon_switches.find(arg) != wincon_switches.end() ||
-			std::find_if(wincon_arg_switches.begin(), wincon_arg_switches.end(), switch_matches_arg) != wincon_arg_switches.end()) {
-			native_console_implied = true;
+		if(terminal_switches.find(arg) != terminal_switches.end() ||
+			std::find_if(terminal_arg_switches.begin(), terminal_arg_switches.end(), switch_matches_arg) != terminal_arg_switches.end()) {
+			terminal_implied = true;
 		}
 
+#ifdef _WIN32
 		if(arg == "--wnoconsole") {
-			native_console_force = false;
+			terminal_force = false;
 		} else if(arg == "--wconsole") {
-			native_console_force = true;
+			terminal_force = true;
 		} else if(arg == "--wnoredirect") {
 			log_redirect = false;
 		}
+#endif
+		if(arg == "--no-log-to-file") {
+			terminal_force = true;
+		} else if(arg == "--log-to-file") {
+			file_force = true;
+		}
 	}
 
-	if(native_console_force.value_or(native_console_implied)) {
+	const bool log_to_terminal = terminal_force.value_or(terminal_implied);
```

Comment:
The variable is misnamed on Windows - on there it's whether to attach a terminal, but IIUC `--wconsole` without `--wnoredirect` should enable a console without cancelling the log redirection.